src/__stories__/carousel-on-different-container-sizes-story.tsx

```diff
@@ -20,29 +20,29 @@ export default {
 type Args = {
     numItems: number;
     itemsPerPageMobile: number;
-    itemsPerPageDesktop: number;
+    itemsPerPageTablet: number;
+    itemsPerPageDesktopSmall: number;
+    itemsPerPageDesktopMedium: number;
+    itemsPerPageDesktopLarge: number;
     withBullets: boolean;
-    nextPageOffset: number;
-    prevPageOffset: number;
-    free: boolean;
-    itemsToScroll: number;
-    autoplay: boolean;
-    loop: boolean;
 };
 
 const ExampleCarousel = ({
+    numItems,
+    withBullets,
     cardsTitlePrefix,
-    intemsPerPage,
+    itemsPerPage,
 }: {
+    numItems: number;
+    withBullets: boolean;
     cardsTitlePrefix: number;
-    intemsPerPage: number;
+    itemsPerPage: {mobile: number; tablet: number; desktop: {small: number; medium: number; large: number}};
```

Comment:
alternative (and backwards compatible) API:
```ts
itemsPerPage: 
    | number 
    | {mobile: number: tablet: number, desktop: number | {small: number; medium: number; large: number} }
```

So all these options would be valid:
```ts
itemsPerPage={5} // applies for all sizes
itemsPerPage={mobile: 3, tablet: 5, desktop: 8}
itemsPerPage={mobile: 3, tablet: 5, desktop: {small: 3, medium: 5: large: 8}}}
```

Reply:
I prefer to support the same than in the carousel component (at least the more similar)